### PR TITLE
PDF Generator initial commit

### PR DESCRIPTION
**Pull request by JoeChapman**
Create a service to run in docker that takes a template param and returns json with a pdf string.
Once this is merged, will transfer to UKHomeoffice

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (easternbloc) [inline .dockerignore]: you could add more to this live LICENSE, README etc

File: index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+app.use((err, req, res, next) => {
+  if (err.code) {
+    res.status(400);
+  } else {
+    res.status(500);
+  }
+  res.json(err);
+});
+
+app.listen(8080);
```
Comment (easternbloc) [inline index.js]: config me up!

File: lib/create-document.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const pdf = require('html-pdf');
+const mustache = require('mustache');
+const phantomPath = require('witch')('phantomjs-prebuilt', 'phantomjs');
+
+module.exports = (req, res, next) => {
+  const options = req.body || {};
+  const data = options.data;
+  let template = options.template;
+
+  if (data) {
```
Comment (easternbloc) [inline lib/create-document.js]: maybe test for template also?

File: lib/create-document.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    // if data, assume mustache template
+    try {
+      template = mustache.render(template, data);
+    } catch (err) {
```
Comment (easternbloc) [inline lib/create-document.js]: this might be another error?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ && yum clean all \
+ && yum install bzip2 -y \
+ && yum install fontconfig -y \
+ && rpm --rebuilddb
+
+COPY package.json /app/package.json
+RUN npm --loglevel warn install --production
+COPY . /app
+
+EXPOSE 8080
```
Comment (easternbloc) [inline Dockerfile]: config

File: README.md [hunk 6]
```diff
@@ -1,2 +1,32 @@
...
+     -i localhost:8080/pdf
+```
+
+HTML
+```bash
+curl -H "Content-Type:application/json" \
+     -d '{
+            "template": "<html><head>My Title </head><body><p>HelloEveryeon</p></body></html>"
+         }' \
+     -i localhost:8080/pdf
```
Comment (easternbloc) [inline README.md]: i'd like this to all become config that can be set via an ENV VAR

File: package.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "pdf-generator",
```
Comment (easternbloc) [inline package.json]: also, i think the name is great but it's not inline with the name in github can you rename the github name. I also think ownership should be in standard UKHomeOffice not Forms

Comment (JoeChapman) [inline package.json]: I agree. I can change the github name and transfer ownership after this is merged.

Comment (easternbloc) [inline package.json]: not sure this should be merged see #2

Comment (easternbloc) [inline package.json]: think a fresh approach should be taken

File: README.md (hunk 6 above)
Comment (JoeChapman) [inline README.md]: this is an example. What do you propose?

**Pull request closed without merging**